scripts/uosc/elements/Menu.lua

```diff
@@ -1065,49 +1065,63 @@ function Menu:render()
+				local clip_ax = ax + icon_size + spacing * 1.5
+				local icon_rect = {ax = rect.ax, ay = rect.ay, bx = clip_ax, by = ay}
+
+				if is_current and requires_submit and get_point_to_rectangle_proximity(cursor, icon_rect) == 0 then
+					cursor.on_primary_down = function() self:search_submit() end
+					icon_opacity = text_opacity
+				end
+
+				ass:icon(rect.ax + spacing + icon_size / 2, rect.cy, icon_size, 'search', {
+					color = fg, opacity = icon_opacity, shadow = 1, shadow_color = bg,
+				})
+
+				-- Query/Placeholder
 				if menu.search.query ~= '' then
-					-- Add a ZWNBSP suffix to prevent libass from trimming any trailing spaces
+					-- Add a ZWNBSP suffix to prevent libass from trimming trailing spaces
 					local query = ass_escape(menu.search.query) .. '\239\187\191'
-					ass:txt(bx - spacing, title_ay + (title_height / 2), 6, query, {
+					ass:txt(rect.bx - spacing, rect.cy, 6, query, {
 						size = self.font_size, color = bgt, wrap = 2, opacity = menu_opacity,
-						clip = '\\clip(' .. ax + spacing .. ',' .. title_ay .. ',' .. bx - spacing .. ',' .. ay .. ')',
+						clip = '\\clip(' .. clip_ax .. ',' .. rect.ay .. ',' .. bx - spacing .. ',' .. ay .. ')',
 					})
 				else
-					ass:txt(bx - spacing, title_ay + (title_height / 2), 6, 'search for...', {
-						size = self.font_size, italic = true, color = bgt, wrap = 2, opacity = menu_opacity * 0.5,
-						clip = '\\clip(' .. ax + spacing .. ',' .. title_ay .. ',' .. bx - spacing .. ',' .. ay .. ')',
+					local placeholder = requires_submit and t('type & ctrl+enter to search') or t('type to search')
+					ass:txt(rect.bx - spacing, rect.cy, 6, placeholder, {
+						size = self.font_size, italic = true, color = bgt, wrap = 2, opacity = menu_opacity * 0.4,
+						clip = '\\clip(' .. ax + spacing .. ',' .. rect.ay .. ',' .. bx - spacing .. ',' .. ay .. ')',
 					})
 				end
+
+				-- Cursor
+				local font_size_half = round(self.font_size / 2)
+				local cursor_ax, cursor_thickness = rect.bx - spacing + 1, round(self.font_size / 14)
+				ass:rect(cursor_ax, rect.cy - font_size_half, cursor_ax + cursor_thickness, rect.cy + font_size_half, {
+					color = fg, opacity = menu_opacity * 0.5
+				})
 			else
 				menu.ass_safe_title = menu.ass_safe_title or ass_escape(menu.title)
-				ass:txt(ax + menu.width / 2, title_ay + (title_height / 2), 5, menu.ass_safe_title, {
-					size = self.font_size, bold = true, color = bg, wrap = 2, opacity = menu_opacity,
-					clip = '\\clip(' .. ax + 2 .. ',' .. title_ay .. ',' .. bx - 2 .. ',' .. ay .. ')',
+				ass:txt(rect.cx, rect.cy, 5, menu.ass_safe_title, {
+					size = self.font_size, bold = true, color = bgt, wrap = 2, opacity = menu_opacity,
+					clip = '\\clip(' .. rect.ax + 2 .. ',' .. rect.ay .. ',' .. rect.bx - 2 .. ',' .. rect.by .. ')',
 				})
 			end
+
+			-- Do nothing when user clicks title
+			if is_current and not cursor.on_primary_down and get_point_to_rectangle_proximity(cursor, rect) then
```

Comment:
You're checking `cursor.on_primary_down` to not overwrite the icon event, but if some element below the menu already set an event we should still overwrite it.

Also maybe clicking the title should start a search?
On the other hand having a place to click while the menu is open might be nice for focusing the window.